de.dlr.sc.overtarget.language.tests/src/de/dlr/sc/overtarget/language/generator/util/ReferencedTargetHelperTest.xtend

```diff
@@ -177,11 +173,54 @@ class ReferencedTargetHelperTest {
 			}".getBytes();
 		val sourceTarget2 = new ByteArrayInputStream(bytes2);
 		tmodelFile2.create(sourceTarget2, IResource.NONE, null);
-		
-		Resource.Factory.Registry.INSTANCE.getExtensionToFactoryMap().put("tmodel", resourceFactory);
-		val testTmodelResource2 = rs.getResource(URI.createPlatformResourceURI("/testProject2/target/target2.tmodel", true), true)
-		val tmodel2 = testTmodelResource2.contents.get(0) as TargetModel
-		
-		Assert.assertNull(refTargetHelper.findTargetfileOfTmodel(tmodel2, outputDirectory, uri))
+
+		Assert.assertNull(refTargetHelper.findTargetFileInProject(tmodelFile2, outputDirectory))
+	}
+	
+	@Test
+	def void testFindTargetForReferencesFile() {
+		val outputDirectory = "./target"
+		val workspace = ResourcesPlugin.getWorkspace();
+		val root = workspace.getRoot();
+		val project  = root.getProject("testProject3");
```

Comment:
Maybe use final varibales here 
`val PROJECT_NAME = "testProject"`

Why these numbers? Maybe better name them according to their distinguishing feature. -> ProjectWithoutTmodel, ProjectWithTmoldel...